token/program-2022/src/instruction.rs

```diff
@@ -1858,6 +1871,31 @@ pub(crate) fn encode_instruction<T: Into<u8>, D: Pod>(
     }
 }
 
+/// Creates a `WithdrawExcessLamports` Instruction
+pub fn withdraw_excess_lamports(
+    token_program_id: &Pubkey,
+    source_account: &Pubkey,
+    authority: &Pubkey,
+    destination_account: &Pubkey,
+    signers: Vec<&Pubkey>,
+) -> Result<Instruction, ProgramError> {
+    let mut accounts = vec![
```

Comment:
Since this is only supported in token-2022, can you add a check, similar to other instructions like

```
check_program_account(token_program_id)?;
```